anitya/tests/db/test_models.py

```diff
@@ -260,6 +360,79 @@ def test_packages__repr__(self):
         self.assertEqual(str(pkg), '<Packages(1, Fedora: geany)>')
 
 
+class ProjectFlagTestCase(DatabaseTestCase):
+    """ Tests for ProjectFlag model. """
+
+    def test_project_flag__repr__(self):
+        """ Test the ProjectFlag.__repr__ function. """
+        flag = create_flagged_project(self.session)
+
+        self.assertEqual(str(flag), '<ProjectFlag(geany, dgay@redhat.com, open)>')
+
+    def test_project_flag__json__(self):
+        """ Test the ProjectFlag.__json__ function. """
+        flag = create_flagged_project(self.session)
+        data = {
+            'created_on': time.mktime(flag.created_on.timetuple()),
+            'user': u'dgay@redhat.com',
+            'state': u'open',
+            'project': u'geany',
+            'updated_on': time.mktime(flag.updated_on.timetuple()),
+            'id': 1
+        }
+
+        self.assertEqual(flag.__json__(), data)
+
+        data['reason'] = u'This is a duplicate.'
+        self.assertEqual(flag.__json__(detailed=True), data)
+
+    def test_project_flag_all(self):
+        """ Test the ProjectFlag.all function. """
+        create_flagged_project(self.session)
+
+        flags = models.ProjectFlag.all(self.session)
```

Comment:
Ooof. I'm fine adding this test, but this function could be deleted and replaced with the normal SQLAlchemy ``models.ProjectFlag.query.all()``.

Reply:
I tried to create tests for methods that weren't covered. This was one of them.

Reply:
Yep, I just wanted to let you know this is not a normal pattern :smile: